### DISCUSSION TITLE
Using a ValueObject does "double" casting?

### DISCUSSION THREAD

**Discussion by parijke** [spatie/laravel-data#840] [General]
I have a VO, that is essentially an extend from CarbonImmutable. In the model we have a cast from
`'date_of_entry' => 'datetime',`

When the field is typed as CarbonImmutable, this works fine, but when it is the DateOfEntry, it fails with 
```
Could not cast date 2024-07-07 00:58:40` into a `App\ValueObject\DateOfEntry` using formats: Y-m-d\TH:i:s, Y-m-d
```

And I do not understand why?

**Comment by rubenvanassche** [comment]
You probably should give the cast another date input format since `2024-07-07 00:58:40` isn't correct as `Y-m-d\TH:i:s` or `Y-m-d`
